xpsparser/parser.py

```diff
@@ -22,42 +22,580 @@
+            
+
+class ProdigyParserXY():
+    """A parser for reading in ASCII-encoded .xy data from Specs Prodigy.
+    
+    Tested with SpecsLab Prodigy v 4.64.1-r88350.
+    """
+    
+    def __init__(self, **kwargs): 
+        """Construct the parser."""
+        
+        self.kwarg_keys = ['author', 'sample', 'experiment_id', 'project',
+                           'axis_id', 'primary_channel_id']
+        
+        self.default_axis_channel_id = 0
+        self.default_primary_channel_id = 1
+        
+        for key in self.kwarg_keys:
+            if key in kwargs.keys():
+                setattr(self, key, kwargs[key])
+            elif key == 'axis_id':
+                setattr(self, key, self.default_axis_channel_id)
+            elif key == 'primary_channel_id':
+                setattr(self, key, self.default_primary_channel_id) 
+
+        self.primary_spectrum_indicators = ['region', 'Region']
+        self.external_channel_indicators = ['external channel', 'External Channel']
+        
+        self.group_metadata_attribute_map = {
+            'Acquisition Date':'timestamp',
+            'Dwell Time':'dwell_time',
+            'Group':'group_name',
+            'Number of Scans':'n_scans',
+            'Region':'spectrum_region',
+            'Excitation Energy':'excitation_energy',
+            'Values/Curve':'n_values',
+            'Source':'source_label',            
+            }
+        
+        self.settings_attribute_map = {
+            'Analysis Method':'analysis_method',
+            'Analyzer Lens':'analyzer_lens',
+            'Analyzer Slit':'analyzer_slit',
+            'Detector Voltage':'detector_voltage',
+            'Eff. Workfunction':'workfunction',
+            'Scan Mode':'scan_mode',
+            }
+        
+        self.default_energy_unit = 'eV'
+        
+        self.known_channel_labels = {'Ring Current':'ring current',
+                                     'I_mirror':'mirror current', 
+                                     'Excitation Energy':'excitation energy',
+                                     'TEY':'total electron yield',
+                                     }
+        
+        self.known_channel_units = {'[mA]':'mA',
+                                    '[V]':'V', 
+                                    '[eV]':'eV',
+                                    }
+        
+        self.known_device_names = {'AMC Mono (TCP)':'monochromator',
+                                   'UE56/2-PGM1 (TCP)':'beamline',
+                                   'ARMIN-ADC3':'armin',
+                                   'Armin10':'armin',
+                                   }
+        
+        self.default_primary_device_name = 'Phoibos Hemispherical Analyzer'
+        self.default_axis_device_name = 'HSA 3500 plus'
+        
+        self.default_method_type = 'XPS'
+        
+        self.precision = 3
+        self.line_nr = 0
+
+    def parseFile(self, filepath, commentprefix = '#'):
```

Comment:
Mark's code does not adhere to our style conventions. Method names need to be `snake_case`.